hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerSizeCountTask.java

```diff
@@ -0,0 +1,177 @@
+
+  private StorageContainerServiceProvider scmClient;
+  private ContainerManager containerManager;
+  private final long interval;
+  private ContainerCountBySizeDao containerCountBySizeDao;
+  private DSLContext dslContext;
+
+
+  public ContainerSizeCountTask(
+      ContainerManager containerManager,
+      StorageContainerServiceProvider scmClient,
+      ReconTaskStatusDao reconTaskStatusDao,
+      ReconTaskConfig reconTaskConfig,
+      ContainerCountBySizeDao containerCountBySizeDao,
+      DSLContext dslContext) {
+    super(reconTaskStatusDao);
+    this.scmClient = scmClient;
+    this.containerManager = containerManager;
+    this.containerCountBySizeDao = containerCountBySizeDao;
+    this.dslContext = dslContext;
+    interval = reconTaskConfig.getContainerSizeCountTaskInterval().toMillis();
+  }
+
+
+  @Override
+  protected synchronized void run() {
+    try {
+      while (canRun()) {
+        wait(interval);
+
+        final List<ContainerInfo> containers = containerManager.getContainers();
+        Map<ContainerSizeCountKey, Long> containerSizeCountMap =
+            new HashMap<>();
+        for (int i = 0; i < containers.size(); i++) {
+          handlePutKeyEvent(containers.get(i), containerSizeCountMap);
+        }
+        // Truncate table before inserting new rows
+        int execute = dslContext.delete(CONTAINER_COUNT_BY_SIZE).execute();
+        LOG.info("Deleted {} records from {}", execute,
+            CONTAINER_COUNT_BY_SIZE);
+
+        writeCountsToDB(true, containerSizeCountMap);
+
+        LOG.info("Completed a 'reprocess' run of ContainerSizeCountTask.");
+      }
+    } catch (Throwable t) {
+      LOG.error("Exception in Container Size Distribution task Thread.", t);
+      if (t instanceof InterruptedException) {
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  /**
+   * Populate DB with the counts of container sizes calculated
+   * using the dao.
+   */
+  private void writeCountsToDB(boolean isDbTruncated,
+                               Map<ContainerSizeCountKey, Long>
+                                   containerSizeCountMap) {
+    List<ContainerCountBySize> insertToDb = new ArrayList<>();
+
+    containerSizeCountMap.keySet().forEach((ContainerSizeCountKey key) -> {
+      ContainerCountBySize newRecord = new ContainerCountBySize();
+      newRecord.setContainerSize(key.containerSizeUpperBound);
+      newRecord.setCount(containerSizeCountMap.get(key));
+      if (!isDbTruncated) {
+        insertToDb.add(newRecord);
+      }
+    });
+    containerCountBySizeDao.insert(insertToDb);
+  }
+
+  @Override
+  public String getTaskName() {
+    return "ContainerSizeCountTask";
+  }
+
+  /**
+   * Calculate and update the count of containers being tracked by
+   * containerSizeCountMap.
+   * Used by reprocess() and process().
+   *
+   * @param containerInfo OmKey being updated for count
+   */
+  private void handlePutKeyEvent(ContainerInfo containerInfo,
```

Comment:
Try to give better name, as there is no event we are capturing with SCM DB sync, here we are taking data from Recon container cache.